src/azul/plugins/metadata/hca/transform.py

```diff
@@ -973,10 +973,20 @@ def _aggregate_date(self) -> MutableJSON:
         entities = api.not_stitched(self.api_bundle.entities.values())
         submission_dates = (e.submission_date for e in entities)
         update_dates = (e.update_date for e in entities)
+        submission_date = min(filter(None, submission_dates), default=None)
+        update_date = max(filter(None, update_dates), default=None)
         return {
             'document_id': str(self.api_bundle.uuid),
-            'submission_date': min(filter(None, submission_dates), default=None),
-            'update_date': max(filter(None, update_dates), default=None)
+            'submission_date': (
+                None
+                if submission_date is None else
+                format_dcp2_datetime(submission_date)
```

Comment:
Probably time for a helper method/function.

What's worrying me is that this wasn't caught in either IT or unit tests.

Before the fixup, we would pass down a document with `datetime` instances or `None`. The `NullableDatetime.to_index` method expects a `str` or None and should have crashed on a `datetime` instance. The only explanation I have is that there were no `datetime` instances in the document, only `None` values. That is disconcerting, too. Any idea what's going on here?

Reply:
https://github.com/DataBiosphere/azul/issues/3416 was just merged into `develop`, and the fixup here is to make this PR compatible with the changes. [One of the changes](https://github.com/DataBiosphere/azul/pull/3419/files) was that `NullableDatetime.to_index` went from expecting an optional `datetime` to an optional `str`.

Reply:
Got it.